### ISSUE TITLE
Loading world in a new thread

### ISSUE THREAD

**Issue by bjubes** [TeamPorcupine/ProjectPorcupine#1593]
Currently loading is all done on the main thread in the Scene via the WorldController. we should load using the same technique as saving (using a Thread that is checked for completion in a coroutine) in order to speed up loading and move towards a completely asynchronous load/save mechanic as per #1459. 

 I have quickly tried to implement this. I have gotten errors, but the overall loadtime dropped from 8 to 4 seconds. Im not sure if this is because the NRE errors are bailing out of code early or not, but regardless this change will add some significant speed to the load time.

_FYI: This change will also speed up development as a large percentage of the time that lapses in between hitting play and the game starting is loading the empty world, NOT compilation. You can see this by loading the main menu instead. the same code is complied, but (for me), Unity takes about 1/4 of the time to begin running_

**Comment by BraedonWooding** [comment]
Yes, I agree this significant.  I think the coroutine should also have an exit state, that way if the user for examples clicks load by accident they can easily cancel and then it will just flush the scene and bring you back to the main menu.  All in all, I think the development speed increase will make its biggest impact now-ish when we are starting to implement new systems that require quick checking of functionality.
